.gitlab-ci.yml

```diff
@@ -125,82 +137,35 @@ build:merge:
     # Runs on tag pipeline where the tag is a prerelease or release version
     - if: $CI_COMMIT_TAG =~ /^v[0-9]+\.[0-9]+\.[0-9]+(?:-.*[0-9]+)?$/
 
-build:linux:
+build:build-generate:
   stage: build
-  needs: []
   script:
+    - mkdir -p ./tmp
     - >
         nix-shell --run '
-        npm run build --verbose;
-        npm test -- --ci --coverage;
+        ./scripts/build-pipelines.sh > ./tmp/build-pipelines.yml
```

Comment:
Technically it's not necessary to run this in the `nix-shell`, as the shell script just runs a simple bash script with no other dependencies.

However in the future, to address more complex binpacking and calculation of the shard count, we can replace the script with a javascript script (or if bin packing is done on the CLI), then it's just safer to run this within the `nix-shell`.